### PR TITLE
Improvements to documentation of platforms

### PR DESCRIPTION
**Pull request by wxtim**
These changes partially address #4342 

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (Tested by building cylc-docs).
- [x] No change log entry required (Doc change).
- [x] This _is_ a documentation update.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -495,14 +495,59 @@
...
+                A list of hosts from which the job host can be selected using
+                :cylc:conf:`[..][selection]method`.
+
+                All hosts should share a file system.
+            ''')
             Conf('job runner', VDR.V_STRING, 'background', desc='''
+
+                .. versionchanged:: 8.0.0
+
+                   This item was formerly known as ``batch system``.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:539]:
```suggestion
                   This item was called ``batch system`` in Cylc 7.
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -495,14 +495,59 @@
...
                 The batch system/job submit method used to run jobs on the
                 platform, e.g., ``background``, ``at``, ``slurm``,
                 ``loadleveler``...
             ''')
-            Conf('job runner command template', VDR.V_STRING)
+            Conf('job runner command template', VDR.V_STRING, desc='''
+                .. versionchanged:: 8.0.0
+
+                   This item was formerly known as
+                   ``batch system command template``.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:549]: The wording is OK, but IMO slightly better to be more direct and explicit as in my previous suggestion for `job runner`.

File: cylc/flow/cfgspec/globalcfg.py [hunk 3]
```diff
@@ -495,14 +495,59 @@
                nedit
         ''')
 
-    with Conf('platforms'):
-        with Conf('<platform name>') as Platform:
+    with Conf('platforms', desc='''
+        .. versionadded:: 8.0.0
```
[the remark this answers is not in the document]
Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:499]: I think that deprecated should be at the top.
I'm less bothered about versionchanged, although I think it's nice to be consistent.

**Review by MetRonnie — APPROVED** [review]

**Comment by wxtim** [comment]
Two reviews.
Addressed all comments from the third.
Documentation.

**Comment by MetRonnie** [comment]
(Remember to let tests run before merging!)

**Pull request merged**